### PR TITLE
MINOR: Improve WorkerUtils#CreateTopics

### PR DESCRIPTION
**Pull request by cmccabe**
Increase the batch size from 10 to 500.  Lengthen the request timeout. Retry failed requests in a background thread even when there are other pending requests.  Do not treat "topic already exists" as an error. Do not fail before the timeout expires.

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java [hunk 1]
```diff
@@ -70,71 +76,129 @@ public static int perSecToPerPeriod(float perSec, long periodMs) {
...
+                                    }
+                                    list.add(newTopicCreation.newTopic.name());
+                                    topics.put(newTopicCreation.newTopic.name(), newTopicCreation.newTopic);
+                                }
+                                for (Map.Entry<String, List<String>> entry : descriptions.entrySet()) {
+                                    if (!entry.getKey().isEmpty()) {
+                                        log.info("Got error {} when attempting to create topic(s): {}",
+                                            entry.getKey(), Utils.join(entry.getValue(), ", "));
+                                    }
+                                }
```
Comment (rajinisivaram) [inline tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java:142]: Can we collect these errors when the createTopics responses are processed below and log them there instead? Perhaps that would simplify the code a bit?

File: tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java [hunk 2]
```diff
@@ -70,71 +76,129 @@ public static int perSecToPerPeriod(float perSec, long periodMs) {
...
+            final Collection<NewTopic> newTopics) throws Throwable {
+        final Properties props = new Properties();
         props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
-        props.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, CREATE_TOPICS_REQUEST_TIMEOUT);
+        props.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, CREATE_TOPICS_REQUEST_TIMEOUT_MS);
+        final ConcurrentLinkedDeque<NewTopicCreation> pendingCreations = new ConcurrentLinkedDeque<>();
+        for (NewTopic newTopic : newTopics) {
+            pendingCreations.offer(new NewTopicCreation(newTopic, ""));
+        }
+        final AtomicInteger remainingCreations = new AtomicInteger(newTopics.size());
```
Comment (rajinisivaram) [inline tools/src/main/java/org/apache/kafka/trogdor/common/WorkerUtils.java:109]: Do we need this variable? We could just check `topicsCreated + topicsVerified`?

**Review by rajinisivaram — COMMENTED** [review]
@cmccabe Thanks for the PR. Looks good, left a couple of minor comments. Also, since there is quite a bit of logic in this method, perhaps it will be good to have unit tests?

**Comment by cmccabe** [comment]
I think @apovzner 's PR #4673 is a simpler fix here.  Will close.

**Pull request closed without merging**